cmd/zed/agents/zfs_retire.c

```diff
@@ -213,16 +270,60 @@ replace_with_spare(fmd_hdl_t *hdl, zpool_handle_t *zhp, nvlist_t *vdev)
 		fmd_hdl_debug(hdl, "zpool_vdev_replace '%s' with spare '%s'",
 		    dev_name, basename(spare_name));
 
-		if (zpool_vdev_attach(zhp, dev_name, spare_name,
-		    replacement, B_TRUE) == 0)
+		ret = zpool_vdev_attach(zhp, dev_name, spare_name,
+		    replacement, B_TRUE);
+		if (ret == 0)
```

Comment:
nit: `ret` is only used in the `if ()` conditional, which is how things current are.  This patch really doesn't need to modify these lines at all.